var/spack/repos/builtin/packages/suite-sparse/package.py

```diff
@@ -65,19 +68,20 @@ def install(self, spec, prefix):
             'CC=%s' % self.compiler.cc,
             'CXX=%s' % self.compiler.cxx,
             'F77=%s' % self.compiler.f77,
-            'CUDA_ROOT     =',
-            'GPU_BLAS_PATH =',
-            'GPU_CONFIG    =',
-            'CUDA_PATH     =',
-            'CUDART_LIB    =',
-            'CUBLAS_LIB    =',
-            'CUDA_INC_PATH =',
-            'NV20          =',
-            'NV30          =',
-            'NV35          =',
-            'NVCC          = echo',
-            'NVCCFLAGS     =',
+            # They "autodetect" CUDA before CUDA_PATH is checked
+            # setting CUDA=no disables them using `which nvcc`,
+            # so that we can use our spack managed cuda path.  If
+            # +cuda we set CUDA_PATH, and the other CUDA specific
+            # libraries are setup by suite-sparse.  See the file
+            # SuiteSparse/SuiteSparse_config/SuiteSparse_config.mk
+            'CUDA=no'
```

Comment:
LOL yeah it seems reasonable that it actually started out as something else, and eventually turned into this.  `CUDA_PATH` is supposed to be an environment variable...but nobody ever sets it or doesn't set it correctly.

I changed it slightly and kept the reference to where to look for future maintainers, but otherwise it's what you suggested (ok I did `format`) :upside_down_face: